umapi_client/functional.py

```diff
@@ -33,8 +33,8 @@ class IdentityTypes(Enum):
 
 
 class GroupTypes(Enum):
-    product = 1
-    # group = product  # deprecated!
+    group = 1
```

Comment:
We can't remove `product` here since it impacts `GroupTypes` validation and will break existing applications that use the client (including the UST).  Let's find a way to issue a deprecation warning if `product` (and perhaps `usergroup` and `productConfiguration`) is used.

Perhaps create a class method on `GroupTypes` called `is_deprecated()` that returns true if the supplied `GroupTypes` variant is deprecated.

Python has a library for handling these types of warnings: https://docs.python.org/3.8/library/warnings.html

The goal is to ensure the warning does not halt the application at all (and can be suppressed using the `warnings` lib if desired).

Reply:
What if I leave 'product' and add 'group' as 4th variable and also make it the default instead of 'product'? Wouldn't this happen:
 -> Old apps/clients that refer to 'product' will still work and will get the deprecation warning directly from UMAPI response,
-> those that use the default will not brake since 'group' should replace all, 
-> and the failing tests... these need review since 'group' should've been used in the first place anyway
Did I get this wrong?

Reply:
We should definitely add `group` to the `GroupTypes` enum and make it the default where applicable. 

Doing that should make those two failing tests pass, but perhaps we ought to make them use `group` instead of `product` as the type.

We should probably add support to the UMAPI client to handle warnings from the UMAPI, but that is outside the scope of this PR.  For now, let's issue our own deprecation warning about using the correct group type.

Reply:
I've applied these changes:
 - kept product in the `GroupTypes` `Enum` so that whomever references it can still use it
 - added `group` as last GroupType and made it default in the two functions: `add_to_groups` and `remove_from_groups` refering to user accounts
 - updated the tests so that the group change reflects all over the test

I've done a local pytest run and the results were:
```
tests/test_actions.py .....................                                                               [ 12%]
tests/test_connections.py ......................................                                    [ 35%]
tests/test_functional.py .............................................................                [ 71%]
tests/test_legacy.py ....................                                                                 [ 83%]
tests/test_live.py ssssssss                                                                            [ 88%]
tests/test_queries.py ....................                                                               [100%]

============= 160 passed, 8 skipped in 78.63s (0:01:18) ============= 
```
 I did not implement any warning client side since the UMAPI should reply with the warning itself (will follow up internally to see if this can be re-added). 
UST side I'm not expecting any errors since whoemever uses it, the .pex file already contains the umapi_client package with the old setting. Also, I did not find anything in the UST source code pointing to 'product' as forced GroupType - therefore, I think it gets whatever the default from umapi_client is.